internal/bits/bit_array.go

```diff
@@ -277,7 +277,10 @@ func (bA *BitArray) getTrueIndices() []int {
 	}
 	// handle last element
 	lastElem := bA.Elems[numElems-1]
-	numFinalBits := bA.Bits - curBit
+	numFinalBits := bA.Bits % 64
```

Comment:
I'm failing to see how it's different than `numFinalBits := bA.Bits - curBit`.

We'll need a test case.

```go
func TestPickRandomCorrectlyHandlesLastElem(t *testing.T) {
	bA := NewBitArray(128)
	bA.SetIndex(127, true)

	elem, ok := bA.PickRandom()
	require.True(t, ok)
	require.Equal(t, 127, elem)
}
```